server_src/runners/state.js

```diff
@@ -977,85 +979,17 @@ WebSocketClient.prototype.onclose = function (e) {
   }
 };
 
-class ClientAPI {
-  static async getRetry(printerURL, apikey, item) {
-    try {
-      logger.info(
-        `Attempting to connect to API: ${item} | ${printerURL} | timeout: ${timeout.apiTimeout}`
-      );
-      const apiConnect = await ClientAPI.get(printerURL, apikey, item);
-      return apiConnect;
-    } catch (err) {
-      logger.error(
-        `Error attempting to connect to API: ${item} | ${printerURL} | timeout: ${timeout.apiTimeout}`,
-        JSON.stringify(err.message)
-      );
-      // If timeout exceeds max cut off then give up... Printer is considered offline.
-      if (timeout.apiTimeout >= timeout.apiRetryCutoff) {
-        logger.info(`Timeout Exceeded: ${item} | ${printerURL}`);
-        // Reset timeout for next printer...
-        timeout.apiTimeout = Number(timeout.apiTimeout) - 9000;
-        throw err;
-      }
-      timeout.apiTimeout += 9000;
-      logger.info(
-        `Attempting to re-connect to API: ${item} | ${printerURL} | timeout: ${timeout.apiTimeout}`
-      );
-      return await ClientAPI.getRetry(printerURL, apikey, item);
-    }
-  }
-
-  static files(printerURL, apikey, item) {
-    const url = `${printerURL}/api/${item}`;
-    fetch(url, {
-      method: "GET",
-      headers: {
-        "Content-Type": "application/json",
-        "X-Api-Key": apikey,
-      },
-    });
-  }
-
-  static post(printerURL, apikey, item, data) {
-    const url = `${printerURL}/api/${item}`;
-    return Promise.race([
-      fetch(url, {
-        method: "POST",
-        headers: {
-          "Content-Type": "application/json",
-          "X-Api-Key": apikey,
-        },
-        body: JSON.stringify(data),
-      }),
-      new Promise((_, reject) =>
-        setTimeout(() => reject(new Error("timeout")), timeout.apiTimeout)
-      ),
-    ]);
-  }
-
-  static get(printerURL, apikey, item) {
-    const url = `${printerURL}/${item}`;
-    return Promise.race([
-      fetch(url, {
-        method: "GET",
-        headers: {
-          "Content-Type": "application/json",
-          "X-Api-Key": apikey,
-        },
-      }),
-      new Promise((_, reject) =>
-        setTimeout(() => reject(new Error("timeout")), timeout.apiTimeout)
-      ),
-    ]);
-  }
-}
-
 class Runner {
+  static octoPrintService = undefined;
+
   static async init() {
     farmPrinters = [];
     const server = await ServerSettings.check();
     systemSettings = server[0];
     timeout = systemSettings.timeout;
```

Comment:
This is the timeout variable I mean. The `OctoprintApiClientService` has a copy of it. Its shared a couple lines below.

Reply:
@davidzwa that's the object from the database. Timeout.apiretry, timeout.websocket yada. Check the server settings modal. I put in a variable for global access at the time due to not using a cache.